File: java/src/main/java/com/up42/challenge/controller/FeatureController.java
package com.up42.challenge.controller;

import com.up42.challenge.model.Feature;
import com.up42.challenge.service.FeatureService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/features")
public class FeatureController {

    private final FeatureService featureService;

    public FeatureController(FeatureService featureService) {
        this.featureService = featureService;
    }

    @GetMapping
    public ResponseEntity<List<Feature>> getAllFeatures() {
        List<Feature> features = featureService.loopExtractLoop();
        return ResponseEntity.ok(features);
    }
}

File: kotlin/src/main/kotlin/com/up42/challenge/controller/FeatureController.kt
package com.up42.challenge.controller


import com.up42.challenge.entity.Feature
import com.up42.challenge.service.FeatureService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class FeatureController(private val featureService: FeatureService) {



    @GetMapping("/features")
    fun getFeatures(): List<Feature> {
        return featureService.loopExtractLoop()
    }
}
